### ISSUE TITLE
Move intermediate console output to worker log files?

### ISSUE THREAD

**Issue by pat-s** [ropensci/drake#909]
In my current setup I create a global log drake log file `drake.log` and log files for the individual workers `worker%a.log`.

```r
drake_config(plan_paper,
             verbose = 2, lazy_load = "promise",
             console_log_file = "log/drake.log", cache_log_file = "log/cache3.log",
             caching = "worker",
             template = list(log_file = "log/worker%a.log", n_cpus = 4, memory = 10000),
             prework = quote(future::plan(future.callr::callr, workers = 4)),
             garbage_collection = TRUE, jobs = 10, parallelism = "clustermq"
)
```

However, the console output of all workers is written to `drake.log` and the worker log only contains information about the R session startup and the final result (when the worker terminates). 

Is is possible to move the intermediate console output to the log files of the workers?
When spawning multiple workers which create similar outputs I cannot relate the output to the individual workers anymore with the current behavior.

**Comment by wlandau** [comment]
I understand the rationale, but I would prefer to keep `drake`'s console log file whole. If you are not looking for worker-specific information, it turns out to be easier than wading through a pile of separate logs. However, `drake.log` should contain the node name and process ID on the left. If you modified your `clustermq` template file to print out this information too, then you could match the `clustermq` logs to `drake.log`.

**Comment by pat-s** [comment]
I understand. While the node name and process ID are useful, it is not possible to trace back to the worker number with only this information.
So in my special case I am running a matrix of 100 targets and I do not know which worker got assigned which target. 
In `drake.log` I see the node name and process ID but since multiple targets are processed on the same node at the same time, I do not know to which target the console output belongs to.

This is critical in the following case:
- Let's say 7 out of 10 targets run fine
- 3 are running without error but it turns out they are running INF or longer than expected (e.g. due to suboptimal argument specs)

-> since 7 out of 10 continuously produce output, finish targets, start new targets and so on, I do not recognize that some targets are actually not making any progress. 

Would it be possible to somehow write out the target name next to the node name and process id? Not sure how easy it is for _drake_ to parse this info into the log.
But that would help a lot in such scenarios to actually track if certain targets show some output at all after time X.

**Comment by wlandau** [comment]
> Would it be possible to somehow write out the target name next to the node name and process id?

Hmm... I thought I already did that:

https://github.com/ropensci/drake/blob/5546cc5c15939467d42d4b36c39e4077c8d5a23a/R/backend-clustermq.R#L164

**Comment by pat-s** [comment]
It might be that we're talking about two different log outputs here:

1. `c0 | 68516 | 2019-06-16 14:12:14.843646 | build bm_vi_task_xgboost_variance on an hpc worker`

2. `c0 | 68669 | 2019-06-16 13:47:01.407176 | [Tune-y] 83: rmse.test.rmse=2.9174478; time: 0.6 min`

Output 2) is not coming from _drake_ but from an arbitrary function within my workflow. Is it also possible to squeeze the target name in these kind of outputs?

I think 1) only pops up during start and end of a target?

**Comment by wlandau** [comment]
Ah, I see what you mean now. Should be fixed in da7d2abc5e1c3486c33197afaf285eaf31af733a.

**Comment by pat-s** [comment]
Works! This helps me a lot. Thanks!! :rocket:
